### PR TITLE
instantiate result type of Apply tree nodes

### PR DESCRIPTION
**Pull request by mprzysucha**

### CODE REVIEW

**Review by sjrd — CHANGES_REQUESTED** [review]
Thank you for the PR. Overall, this looks good. I have left a couple comments to better adhere to our conventions.

You may directly `--amend` and force-push your commit with any follow-up changes.

File: tasty-query/shared/src/test/scala/tastyquery/TypeSuite.scala [hunk 1]
```diff
@@ -44,6 +44,14 @@ class TypeSuite extends UnrestrictedUnpicklingSuite {
     def isArrayOf(arg: Type => Boolean)(using Context): Boolean =
       isApplied(_.isRef(defn.ArrayClass), Seq(arg))
 
+  testWithContext("apply-dependent") {
+    val DependentMethodClass = ctx.findTopLevelClass("simple_trees.DependentMethod")
+    val testVal = DependentMethodClass.findNonOverloadedDecl(name"test")
+    val testDef = testVal.tree.get.asInstanceOf[ValDef]
+    val applyTree = testDef.rhs.get.asInstanceOf[Apply]
+    assert(applyTree.tpe.isSubtype(defn.StringType))
```
[the remark this answers is not in the document]
Comment (mprzysucha) [reply tasty-query/shared/src/test/scala/tastyquery/TypeSuite.scala:52]: If I correctly understand I can use here `isOfClass` with String `ClassSymbol`, where `isOfClass` checks `underlying` Type for `ConstantType`.

File: tasty-query/shared/src/main/scala/tastyquery/Trees.scala [hunk 2]
```diff
@@ -310,7 +310,7 @@ object Trees {
       funTpe.widenOverloads match
         case funTpe: MethodType =>
           // TODO: substitute parameters when dependent
```
[the remark this answers is not in the document]
Comment (mprzysucha) [reply tasty-query/shared/src/main/scala/tastyquery/Trees.scala]: done :)

File: tasty-query/shared/src/test/scala/tastyquery/TypeSuite.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (sjrd) [reply tasty-query/shared/src/test/scala/tastyquery/TypeSuite.scala]: Yes, that should work. You can get the String `ClassSymbol` with `defn.StringClass`.

[the remark this answers is not in the document]
Comment (mprzysucha) [reply tasty-query/shared/src/test/scala/tastyquery/TypeSuite.scala]: yes, I took exactly `defn.StringClass`, thank you, it is pushed

**Review by sjrd — APPROVED** [review]

**Pull request merged**